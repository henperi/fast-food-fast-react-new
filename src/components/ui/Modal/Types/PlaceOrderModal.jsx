import React from 'react';
import { connect } from 'react-redux';
import placeOrder from '../../../../actions/orderActions';

const PlaceOrderModal = props => (
  <div>
    <h2 className="text-center">You are about to place an order</h2>
    <form onSubmit={() => {}} method="POST" className=" card card-shadow">
      <div className="">
        <h3>Are you sure you want to order all the items in cart?</h3>
        <button
          onClick={() => {
            props.placeOrder(props.cart, props.handleModal);
            // alert('aa');
          }}
          type="button"
          className="btn btn-green btn-rounded btn-bg close-button "
        >
          Proceed
        </button>
        <button
          onClick={props.handleModal}
          type="button"
          className="close-button btn btn-primary btn-rounded btn-bg"
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
);

const mapStateToProps = state => ({
  cart: state.cart,
});

export default connect(
  mapStateToProps,
  { placeOrder },
)(PlaceOrderModal);
