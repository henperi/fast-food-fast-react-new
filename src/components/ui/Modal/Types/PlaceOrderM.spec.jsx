/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { PlaceOrderModal } from './PlaceOrderModal';

describe('rendering', () => {
  let wrapper;
  let props;
  // let mockedSignUpState;
  beforeEach(() => {
    props = {
      handleModal: jest.fn(),
      cart: [],
      placeOrder: jest.fn(),
    };
  });

  it('should render the component', () => {
    wrapper = shallow(<PlaceOrderModal {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });

  it('should test the placeOrder function', () => {
    const fakeEvent = { preventDefault: () => console.log('preventDefault') };
    wrapper = shallow(<PlaceOrderModal {...props} />);
    const placeOrderBtn = wrapper.find('.btn-green');
    placeOrderBtn.simulate('click', fakeEvent);
    const placeOrder = jest.fn(props.cart);
    const promise = new Promise((resolve) => {
      resolve(wrapper.instance().onSubmit);
    });
    promise.then(() => expect(placeOrder).toHaveBeenCalledTimes(1));
  });

  it('should test the close-modal btn click', () => {
    const fakeEvent = { preventDefault: () => console.log('preventDefault') };
    wrapper = shallow(<PlaceOrderModal {...props} />);
    const closeBtn = wrapper.find('.btn-primary');
    closeBtn.simulate('click', fakeEvent);
    const handleModal = jest.fn(props.cart);
    const promise = new Promise((resolve) => {
      resolve(wrapper.instance().onSubmit);
    });
    promise.then(() => expect(handleModal).toHaveBeenCalledTimes(1));
  });
});
