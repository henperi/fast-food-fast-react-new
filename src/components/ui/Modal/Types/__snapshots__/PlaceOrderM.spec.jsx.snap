// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
<div>
  <h2
    className="text-center"
  >
    You are about to place an order
  </h2>
  <form
    className=" card card-shadow"
    method="POST"
  >
    <div
      className=""
    >
      <h3>
        Are you sure you want to order all the items in cart?
      </h3>
      <button
        className="btn btn-green btn-rounded btn-bg close-button "
        onClick={[Function]}
        type="button"
      >
        Proceed
      </button>
      <button
        className="close-button btn btn-primary btn-rounded btn-bg"
        onClick={[MockFunction]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
